web/src/app/lists/QueryList.tsx

```diff
@@ -100,55 +103,55 @@ export default function QueryList(props: {
   }
 
   if (noSearch) {
-    delete variables.input.search
+    delete queryVariables.input.search
   }
 
   const { data, loading, fetchMore, stopPolling } = useQuery(aliasedQuery, {
     client: GraphQLClientWithErrors,
-    variables,
+    variables: queryVariables,
     fetchPolicy: 'network-only',
     pollInterval: POLL_INTERVAL,
   })
 
-  let items = []
+  const nodes = data?.data?.nodes ?? []
+  const items = nodes.map(mapDataNode)
   let loadMore
 
-  if (data && data.data && data.data.nodes) {
-    items = data.data.nodes.map(props.mapDataNode)
-    if (data.data.pageInfo.hasNextPage) {
-      loadMore = buildFetchMore(
-        fetchMore,
-        data.data.pageInfo.endCursor,
-        stopPolling,
-      )
-    }
+  if (data?.data?.pageInfo?.hasNextPage) {
+    loadMore = buildFetchMore(
+      fetchMore,
+      data.data.pageInfo.endCursor,
+      stopPolling,
+      queryVariables.input.first,
+    )
   }
 
-  return (
-    <Grid container spacing={2}>
-      {/* Such that filtering/searching isn't re-rendered with the page content */}
-      <Grid item xs={12} className={classes.searchGridItem}>
-        <Search endAdornment={searchAdornment} />
-      </Grid>
-
-      <Grid item xs={12}>
-        <PaginatedList
+  if (Boolean(props.checkboxActions) || !props.noSearch) {
+    return (
+      <Grid container spacing={2}>
+        <ControlledPaginatedList
           {...listProps}
-          key={urlKey}
           items={items}
+          itemsPerPage={queryVariables.input.first}
           loadMore={loadMore}
           isLoading={!data && loading}
         />
       </Grid>
-    </Grid>
-  )
-}
-
-QueryList.defaultProps = {
-  mapDataNode: (n: ObjectMap) => ({
-    title: n.name,
-    url: n.id,
-    subText: n.description,
-  }),
-  variables: {},
+    )
+  } else {
```

Comment:
No need for `else` since we're returning from the `if`, eslint should have caught it though.

Reply:
eslint hasn't been working properly since we switched to typescript for me (in make check/yarn fmt as well)